### DISCUSSION TITLE
Use of async lock

### DISCUSSION THREAD

**Discussion by levimatheri** [ligershark/dotnet-new-ui#99] [General]
@RehanSaeed I was just reading through the code to familiarize myself, and was wondering the design reason behind adding `AsyncLock` in the [`DotnetCli.RunAsync()` ](https://github.com/ligershark/dotnet-new-ui/commit/931bb71c4e6ed62a2bb2a1e8194cad7954521b52#diff-bbaf5c270d5cbb3cfb6342d903f4a64bcb275ee72928abd427442631e311e15b)

**Comment by RehanSaeed** [comment]
The dotnet exe sometimes does strange things if you execute it multiple times in parallel in my experience with `dotnet new` in particular.

**Reply by levimatheri** [reply]
Gotcha thanks for clarifying
